src/subscription_manager/managercli.py

```diff
@@ -744,6 +744,10 @@ def _do_command(self):
 
         try:
             self._validate_options()
```

Comment:
I found out that since this calls `self.assert_should_be_registered()`, we never reach the code block you added (explains why we hit that earlier). I think it would make since to move your logic about the validity of the identity to that method. This would also cover more cases.